src/external-repository/scripts/clone.sh

```diff
@@ -6,6 +6,21 @@ cd "$(dirname "$0")"
 # Load the variables
 source ./variables.sh
 
+# Function to obtain OIDC token and authenticate with Azure
+obtain_oidc_token_and_authenticate() {
+    if [[ -n "${EXT_GIT_AZURE_CLIENT_ID}" && -n "${EXT_GIT_AZURE_TENANT_ID}" ]]; then
+        echo "Obtaining OIDC token for Azure authentication..."
+        response=$(curl -H "Authorization: bearer $ACTIONS_ID_TOKEN_REQUEST_TOKEN" "$ACTIONS_ID_TOKEN_REQUEST_URL&audience=api://AzureADTokenExchange")
+        federatedToken=$(echo $response | jq -r '.value')
+
+        echo "Logging in to Azure with OIDC token..."
+        az login --service-principal -u $AZURE_CLIENT_ID --tenant $AZURE_TENANT_ID --federated-token "$federatedToken"
```

Comment:
```suggestion
        az login --service-principal -u $AZURE_CLIENT_ID --tenant $AZURE_TENANT_ID --federated-token "$federatedToken" --allow-no-subscriptions
```
You need this parameter if the MI isn't given access to any Azure subscriptions in the tenant.